### PR TITLE
Database semconv stability migration guide

### LINKED ISSUE #1022: Database: write migration guide for (to-be) stable semconv
Similar to https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/migration-guide.md

### PR DESCRIPTION
**Pull request by trask**
Fixes #1022

## Changes

Documents the changes between experimental database semantic conventions and the (to-be) stable database semantic conventions.

### CODE REVIEW

File: docs/database/migration-guide.md
Comment (lmolkova) [inline docs/database/migration-guide.md:52]: nit: 

maybe 
```suggestion
| `db.query.parameter.<key>`                      | New, Opt-In                                                                                                      |
```

to be consistent with 

| `db.connection_string` | Removed

?

File: docs/database/migration-guide.md
Comment (lmolkova) [inline docs/database/migration-guide.md:1]: wonder if we should put migration guides to `docs/{area}/supplementary_guidelines/`  folder - this is (now) a recommended pattern in the spec (per signal) - https://github.com/open-telemetry/opentelemetry-specification/issues/3587

**Comment by trask** [comment]
TODO: add section to document changes to experimental metrics also

**Comment by maryliag** [comment]
> TODO: add section to document changes to experimental metrics also

example: `db.client.connections.usage` -> `db.client.connection.count`

File: docs/database/migration-guide.md
Comment (trask) [reply docs/database/migration-guide.md:52]: the logic(?) behind the current format is that the column normally starts with the old name, but in this case there's no old name, so I put "New: " in front of it

File: docs/http/supplementary-guidelines/migration-guide.md
Comment (trask) [inline docs/http/supplementary-guidelines/migration-guide.md:1]: moved http migration guide to `supplementary-guidelines` directory as well

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (heyams) [inline docs/database/supplementary-guidelines/migration-guide.md:43]: `db.redis.database_index` is `long` data type. whereas `db.namespace` is a `string` type. 

should we mark `db.redis.database_index` as removed instead? or can we add some comments to explain the cross-data type replacement?

Comment (heyams) [inline docs/database/supplementary-guidelines/migration-guide.md:43]:
```suggestion
| `db.redis.database_index` &rarr; `db.namespace`     |  Removed, replaced by `db.namespace`                                                                                                           |
```

Comment (heyams) [reply docs/database/supplementary-guidelines/migration-guide.md:43]: Redis lettuce API `getDatabase` is returning the Redis database number. What value is expected for the new db.namespace for reids? 

https://lettuce.io/core/release/api/io/lettuce/core/RedisURI.html#PARAMETER_NAME_DATABASE:~:text=public%C2%A0int-,getDatabase,-()

Comment (heyams) [reply docs/database/supplementary-guidelines/migration-guide.md:43]: `redisUri.getHost()` as namespace?

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (trask) [reply docs/database/supplementary-guidelines/migration-guide.md]: clarified text

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (trask) [reply docs/database/supplementary-guidelines/migration-guide.md]: I just sent #1233 to clarify, let me know if that helps

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (trask) [inline docs/database/supplementary-guidelines/migration-guide.md:43]: @heyams is this good now? thanks

**Comment by trask** [comment]
I'm thinking we should wait to merge this until after stability, just to avoid confusion in case there are any last-minute changes.

But please review it and leave comments / suggestions / questions.

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (maryliag) [inline docs/database/supplementary-guidelines/migration-guide.md:68]: tiny nit: add a period at the end of this sentence or remove the ones from the bullet points above to keep consistent

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (maryliag) [inline docs/database/supplementary-guidelines/migration-guide.md:83]: feels like the sentence got cut without being finished

File: docs/http/migration-guide.md
Comment (lmolkova) [inline docs/http/migration-guide.md:1]: oops?

Comment (trask) [reply docs/http/migration-guide.md:1]: 😱

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (maryliag) [inline docs/database/supplementary-guidelines/migration-guide.md:158]: On the PR https://github.com/open-telemetry/semantic-conventions/pull/1290 I changed the description from `cumulative` to `current`. Do you also want to add this change here or is that to much level of details and not necessary here?
Up to you!

**Comment by TylerHelmuth** [comment]
Saw that Database semconv was getting ready to mark stable and I love that this has existed along side it ❤️

File: docs/database/supplementary-guidelines/migration-guide.md
Comment (trask) [reply docs/database/supplementary-guidelines/migration-guide.md]: I saw that was more of a clarification since in general cumulative vs delta is a backend concern

**Comment by trask** [comment]
@open-telemetry/specs-semconv-maintainers the policy violations seem unrelated to this PR:

```
Diagnostic report:

Violation: Resource 'service' cannot remove required/recommended attributes (missing '{"service.name", "service.version"}')
  - Category         : backward_compatibility
  - Type             : semconv_attribute
  - SemConv group    : service
  - SemConv attribute: 
  - Provenance: /home/weaver/source

Violation: Resource 'telemetry.sdk' cannot remove required/recommended attributes (missing '{"telemetry.sdk.language", "telemetry.sdk.name", "telemetry.sdk.version"}')
  - Category         : backward_compatibility
  - Type             : semconv_attribute
  - SemConv group    : telemetry.sdk
  - SemConv attribute: 
  - Provenance: /home/weaver/source
```

**Review by lmolkova — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**
